File: gaiag/examples/alarm.py
# Gaiag --- Guile in Asd In Asd in Guile.
# Copyright © 2014, 2015, 2016 Jan Nieuwenhuizen <janneke@gnu.org>
#
# This file is part of Gaiag.
#
# Gaiag is free software: you can redistribute it and/or modify it
# under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
#
# Gaiag is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public
# License along with Gaiag.  If not, see <http://www.gnu.org/licenses/>.
# 
# Commentary:
# 
# Code:

#! /usr/bin/python

import sys
import os
sys.path.insert (0, os.path.dirname (sys.argv[0]))
#
import dzn.AlarmSystem
import locator
import runtime

def detected ():
    sys.stderr.write ('Console.detected\n')

def deactivated ():
    sys.stderr.write ('Console.deactivated\n')

def main ():
    loc = locator.Locator ()
    rt = runtime.Runtime ()
    sut = dzn.AlarmSystem (loc.set (rt), name='alarmsystem')
    sut.console.outport.name = 'console'
    sut.console.outport.self = sut

    sut.console.outport.detected = detected
    sut.console.outport.deactivated = deactivated

    sut.console.inport.arm ()
    sut.sensor.sensor.outport.triggered ()
    rt.flush (sut.sensor)
    sut.console.inport.disarm ()
    sut.sensor.sensor.outport.disabled ()
    rt.flush (sut.sensor)

if __name__ == '__main__':
    main ()

File: gaiag/examples/alarm.c
// Dezyne --- Dezyne command line tools
// Copyright © 2015, 2016 Jan Nieuwenhuizen <janneke@gnu.org>
// Copyright © 2015 Paul Hoogendijk <paul.hoogendijk@verum.com>
//
// This file is part of Dezyne.
//
// Dezyne is free software: you can redistribute it and/or modify it
// under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// Dezyne is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public
// License along with Dezyne.  If not, see <http://www.gnu.org/licenses/>.
//
// Commentary:
//
// Code:

#include "AlarmSystem.h"

#include <dzn/locator.h>
#include <dzn/runtime.h>

static void detected()
{
  DZN_TRACE("Console.detected");
}

static void deactivated()
{
  DZN_TRACE("Console.deactivated");
}

int main()
{
  runtime dezyne_runtime;
  runtime_init (&dezyne_runtime);

  locator dezyne_locator;
  locator_init (&dezyne_locator, &dezyne_runtime);

  AlarmSystem alarmsystem;
  dzn_meta_t m = {"alarmsystem", 0};
  AlarmSystem_init(&alarmsystem, &dezyne_locator, &m);
  alarmsystem.console->out.name = "console";
  alarmsystem.console->out.self = &alarmsystem;

  alarmsystem.console->out.detected = detected;

  alarmsystem.console->out.deactivated = deactivated;

  alarmsystem.console->in.arm(alarmsystem.console);
  alarmsystem.sensor.sensor->out.triggered(alarmsystem.sensor.sensor);
  runtime_flush(&alarmsystem.sensor.dzn_info);
  alarmsystem.console->in.disarm(alarmsystem.console);
  alarmsystem.sensor.sensor->out.disabled(alarmsystem.sensor.sensor);
  runtime_flush(&alarmsystem.sensor.dzn_info);
  return 0;
}
